### PR TITLE
expand ~/ in config filename

### PR DESCRIPTION
**Pull request by parkr**
Expand `~/` in configuration filenames.

### CODE REVIEW

File: datadog.go [hunk 1]
```diff
@@ -22,6 +25,18 @@ func init() {
 	log.SetOutput(os.Stderr)
 }
 
+func expandPath(path string) string {
```
Comment (abhiyerra) [inline datadog.go:28]: Please use this: http://golang.org/pkg/path/filepath/#Abs

Comment (parkr) [inline datadog.go:28]: @abhiyerra That doesn't seem to expand the `~/` shortcut. http://play.golang.org/p/cojcjJJiLt

Comment (abhiyerra) [inline datadog.go:28]: Actually I just realized why do we need to expand the path in the first place? The posix layer should convert that automatically when run.

Comment (parkr) [inline datadog.go:28]: Using GNU Bash-3.2 on my Mac, that automatic expansion doesn't happen.

``` text
~/vsco/datadog#master$ ./datadog -conf=~/.datadogrc
2015/07/01 10:54:56 Config file path from flag: ~/.datadogrc
2015/07/01 10:54:56 error: not enough arguments. usage: datadog TYPE METRIC VALUE(S)...
```

**Pull request merged**